erpnext/selling/doctype/sales_order/sales_order.js

```diff
@@ -242,20 +242,19 @@ erpnext.selling.SalesOrderController = erpnext.selling.SellingController.extend(
 
 							var selected_data = {};
 							var item_data = [];
-							for (i = 0; i < data["items"].length; i++) {
-								for (j = 0; j < visible_items.length; j++) {
-									if (data.items[i].sales_order_item == visible_items[j]) {
+
+							data["items"]
+								.filter(item => { return visible_items.indexOf(item.sales_order_item)>=0; }, visible_items)
```

Comment:
The second argument is not necessary here. That changes the scope of the  filter function which you are not using.